### PR TITLE
Added sending confirmation email after subscription creation or renewal

### PR DESCRIPTION
**Pull request by PiotrWachulec**

### CODE REVIEW

File: src/Modules/Payments/Application/Subscriptions/CreateSubscription/SubscriptionCreatedEnqueueEmailConfirmationHandler.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+            const string sql = "SELECT " +
+                               "[Payer].[Id], " +
+                               "[Payer].[Login], " +
+                               "[Payer].[Email], " +
+                               "[Payer].[FirstName], " +
+                               "[Payer].[LastName], " +
+                               "[Payer].[Name] " +
+                               "FROM [payments].[Payers] AS [Payer] " +
+                               "WHERE [Payer].[Id] = @PayerId";
```
Comment (kgrzybek) [inline src/Modules/Payments/Application/Subscriptions/CreateSubscription/SubscriptionCreatedEnqueueEmailConfirmationHandler.cs:49]: Return only the Email - this is what you only need :).

File: src/Modules/Payments/Application/Subscriptions/RenewSubscription/SubscriptionRenewedEnqueueEmailConfirmationHandler.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+            const string sql = "SELECT " +
+                               "[Payer].[Id], " +
+                               "[Payer].[Login], " +
+                               "[Payer].[Email], " +
+                               "[Payer].[FirstName], " +
+                               "[Payer].[LastName], " +
+                               "[Payer].[Name] " +
+                               "FROM [payments].[Payers] AS [Payer] " +
+                               "WHERE [Payer].[Id] = @PayerId";
```
Comment (kgrzybek) [inline src/Modules/Payments/Application/Subscriptions/RenewSubscription/SubscriptionRenewedEnqueueEmailConfirmationHandler.cs:49]: The same as above - you need only the  email.

 Maybe create some static class PayerEmailProvider with static method which gets email based on id and use it in that 2 places (DRY)?

File: src/Modules/Payments/Application/Subscriptions/CreateSubscription/SubscriptionCreatedEnqueueEmailConfirmationHandler.cs (hunk 1 above)
Comment (PiotrWachulec) [reply src/Modules/Payments/Application/Subscriptions/CreateSubscription/SubscriptionCreatedEnqueueEmailConfirmationHandler.cs:49]: You're right :) I took it from GetPayerQuery handler, but it should be more specialized :)

File: src/Modules/Payments/Application/Subscriptions/RenewSubscription/SubscriptionRenewedEnqueueEmailConfirmationHandler.cs (hunk 2 above)
Comment (PiotrWachulec) [reply src/Modules/Payments/Application/Subscriptions/RenewSubscription/SubscriptionRenewedEnqueueEmailConfirmationHandler.cs:49]: Yup, good idea! Where such class should be placed?

Comment (kgrzybek) [reply src/Modules/Payments/Application/Subscriptions/RenewSubscription/SubscriptionRenewedEnqueueEmailConfirmationHandler.cs:49]: Maybe `Application/Payers` ?

Comment (PiotrWachulec) [reply src/Modules/Payments/Application/Subscriptions/RenewSubscription/SubscriptionRenewedEnqueueEmailConfirmationHandler.cs]: @kgrzybek please look at updated code :)

File: src/Modules/Payments/Application/Subscriptions/CreateSubscription/SubscriptionCreatedEnqueueEmailConfirmationHandler.cs (hunk 1 above)
Comment (PiotrWachulec) [reply src/Modules/Payments/Application/Subscriptions/CreateSubscription/SubscriptionCreatedEnqueueEmailConfirmationHandler.cs]: @kgrzybek please look at updated code :)

File: src/Modules/Payments/Application/Subscriptions/RenewSubscription/SubscriptionRenewedEnqueueEmailConfirmationHandler.cs [hunk 3]
```diff
@@ -26,32 +27,14 @@ public class SubscriptionRenewedEnqueueEmailConfirmationHandler : INotificationH
         
         public async Task Handle(SubscriptionRenewedNotification notification, CancellationToken cancellationToken)
         {
-            var payer = await GetPayerDetails(notification.DomainEvent.PayerId);
+            PayerEmailProvider.Init(_sqlConnectionFactory);
+            
+            var payer = await PayerEmailProvider.GetPayerDetails(notification.DomainEvent.PayerId);
```
Comment (kgrzybek) [inline src/Modules/Payments/Application/Subscriptions/RenewSubscription/SubscriptionRenewedEnqueueEmailConfirmationHandler.cs:32]: This is not a good solution, it introduces temporal coupling https://blog.ploeh.dk/2011/05/24/DesignSmellTemporalCoupling/

Just pass `_sqlConnectionFactory` as an argument to `PayerEmailProvider.GetPayerDetails` method :)

Comment (kgrzybek) [reply src/Modules/Payments/Application/Subscriptions/RenewSubscription/SubscriptionRenewedEnqueueEmailConfirmationHandler.cs:32]: And the method should return string and name should be GetPayerEmail

Comment (PiotrWachulec) [reply src/Modules/Payments/Application/Subscriptions/RenewSubscription/SubscriptionRenewedEnqueueEmailConfirmationHandler.cs]: I think it's done ;)

File: src/Modules/Payments/Application/Payers/GetPayerEmail/PayerEmailProvider.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+namespace CompanyName.MyMeetings.Modules.Payments.Application.Payers.GetPayerEmail
+{
+    public static class PayerEmailProvider
+    {
+        public static async Task<string> GetPayerEmail(Guid payerId, ISqlConnectionFactory sqlConnectionFactory)
+        {
+            var connection = sqlConnectionFactory.GetOpenConnection();
+
+            const string sql = "SELECT " +
+                               "[Payer].[Email], " +
```
Comment (kgrzybek) [inline src/Modules/Payments/Application/Payers/GetPayerEmail/PayerEmailProvider.cs:15]: It won't work, there is a comma at the end, you will have a SQL syntax error.

Comment (PiotrWachulec) [reply src/Modules/Payments/Application/Payers/GetPayerEmail/PayerEmailProvider.cs]: Yup, you're right. Fixed and tested.

**Pull request merged**